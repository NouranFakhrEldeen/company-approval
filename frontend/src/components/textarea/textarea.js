/* eslint-disable no-unused-vars */
import React, { forwardRef } from 'react';
import { capitalizeFirstWord } from '../../helpers';
import './textarea.scss';

const textarea = forwardRef(({
  placeholder,
  value,
  id,
  name,
  classes,
  onChange,
  disabled },
ref,
) => {
  return (
    <div className={'col-sm-5 col-lg-12 padding-0 '}>
      <textarea
        ref={ref}

        data-testid='textArea'
        name={name}
        value={value}
        onChange={(e) => onChange(e)}
        className={classes}
        placeholder={capitalizeFirstWord(placeholder)}
        disabled= {disabled}
      />
    </div>
  );
}
);

export const Textarea = textarea;
