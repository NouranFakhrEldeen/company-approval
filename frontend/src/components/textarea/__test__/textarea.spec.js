import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
// import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Textarea } from '../textarea';

let onChange = jest.fn();

describe('testing textArea', ()=>{
  afterEach(cleanup);


  test('should write in input', () => {
    const { getByTestId } = render(<Textarea data-testid='textArea' onChange={()=> onChange()}/>);
    const input = getByTestId('textArea');
    fireEvent.change(input, { target: { value: '23' } });
    expect(input.value).toBe('23');
  });

  //   snapshot
  test('should do snapshot', () => {
    const textarea = renderer.create(
      <Textarea
        id = {'test'}
        onChange={()=> onChange()}
      />
    ).toJSON();
    expect(textarea).toMatchSnapshot();
  });

});