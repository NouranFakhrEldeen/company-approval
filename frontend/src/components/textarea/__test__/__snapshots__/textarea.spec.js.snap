// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing textArea should do snapshot 1`] = `
<div
  className="col-sm-5 col-lg-12 padding-0 "
>
  <textarea
    data-testid="textArea"
    onChange={[Function]}
  />
</div>
`;
